core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerExoPlayerCreator.java

```diff
@@ -85,7 +85,6 @@ ExoPlayerTwoImpl create(Context context,
                     userAgent,
                     handler,
                     dataSourceFactory,
-                    adsLoader,
```

Comment:
Is the `advertsLoader` still needed as a parameter of the constructor?

Reply:
(I can't see why on the diff)

Reply:
I am talking about line `74`

Reply:
Yes, the only change is that before we were passing `Optional<AdsLoader>` to both `MediaSourceFactory` and `ExoPlayerFacade` constructors in this method. Now it's only passed to the latter since it doesn't need to be a field in the `MediaSourceFactory`

Reply:
Yes. This class is used to create everything related to `no-player`, just the `exo-player` variant. In this case, the `AdsLoader` is passed to the `ExoPlayerFacade` on line 101. The ads loader is then passed to the `MediaSourceFactory` when loading the video, rather than through the constructors.